src/Aspire.Hosting.Docker/ResourceComparer.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#pragma warning disable ASPIRECOMPUTE001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting.Docker;
+
+internal sealed class ResourceComparer : IEqualityComparer<IResource>
```

Comment:
This made me realize we have the same problem with the ACA logic, putting up a fix for that now.

Reply:
See https://github.com/dotnet/aspire/pull/9225/files